### ISSUE TITLE
How to use wildcards in routeRules.

### ISSUE THREAD

**Issue by TangsWang** [nuxt/nuxt#23190]
### Environment

routeRules: {
    '/list/(.*).html?page=(.*)': { redirect: '/tags/$1/$2' },
  }

### Reproduction

routeRules: {
    '/list/(.*).html?page=(.*)': { redirect: '/tags/$1/$2' },
  }

### Describe the bug

routeRules: {
    '/list/(.*).html?page=(.*)': { redirect: '/tags/$1/$2' },
  }

### Additional context

routeRules: {
    '/list/(.*).html?page=(.*)': { redirect: '/tags/$1/$2' },
  }

**Comment by manniL** [comment]
Right now this is not possible via routeRules, only with a custom nitro middleware + sendRedirect.

A first step would be when https://github.com/unjs/nitro/issues/1748 was resolved

**Comment by xjccc** [comment]
could support matcher?
```
routeRules: {
  '/xxx/c6_s93_([A-Z]+)': {
    redirect: ([, a]) => `/xxx/c11_s15_${a}`
  }
}
```

**Comment by manniL** [comment]
@xjccc no, the content should be serializable, hence no functions are allowed in the config. But you could use e.g. $1, $2, ...

That'd be a discussion for https://github.com/unjs/nitro/issues/1748 though and not here.
